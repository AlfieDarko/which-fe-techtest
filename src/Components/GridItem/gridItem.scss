.grid-item {
  height: 200px;
  width: 200px;
  border: 3px solid blue;
  cursor: pointer;

  &--state-highlighted {
    background: lightcoral;
  }
}
