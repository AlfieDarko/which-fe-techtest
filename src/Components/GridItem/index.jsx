import './gridItem.scss'

import React from 'react'
import cx from 'classnames'

const GridItem = ({ number, setClickedItemValue, isCommonMultiple }) => {
  const onClickHandler = () => {
    setClickedItemValue(number)
  }

  const gridItemClassName = cx({ 'grid-item': !isCommonMultiple }, { 'grid-item--state-highlighted': isCommonMultiple })

  return (
    <button type="button" className={gridItemClassName} value={number} data-testid={`grid-item-${number}`} onClick={() => onClickHandler()}>
      <p data-testid="grid-item-value">{number}</p>
    </button>
  )
}

export default GridItem
