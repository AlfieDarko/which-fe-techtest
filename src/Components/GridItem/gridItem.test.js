import { render, screen } from '@testing-library/react'
import React from 'react'

import GridItem from '.'

test('should render correctly', () => {
  const { asFragment } = render(<GridItem number={1}/>)
  expect(asFragment()).toMatchSnapshot()
})

test('should render a grid item', () => {
  render(<GridItem number={1}/>)
  const linkElement = screen.getByTestId('grid-item-1')
  expect(linkElement).toBeInTheDocument()
})

test('should render a grid item with a value', () => {
  render(<GridItem number={2}/>)
  const linkElement = screen.getByTestId('grid-item-value')
  expect(linkElement).toBeInTheDocument()
})

test('should render the grid-item correctly given the correct props', () => {
  const { asFragment } = render(<GridItem number={1} isCommonMultiple/>)
  expect(asFragment()).toMatchSnapshot()
})
