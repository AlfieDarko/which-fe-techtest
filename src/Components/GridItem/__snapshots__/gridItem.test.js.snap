// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <button
    class="grid-item"
    data-testid="grid-item-1"
    type="button"
    value="1"
  >
    <p
      data-testid="grid-item-value"
    >
      1
    </p>
  </button>
</DocumentFragment>
`;

exports[`should render the grid-item correctly given the correct props 1`] = `
<DocumentFragment>
  <button
    class="grid-item--state-highlighted"
    data-testid="grid-item-1"
    type="button"
    value="1"
  >
    <p
      data-testid="grid-item-value"
    >
      1
    </p>
  </button>
</DocumentFragment>
`;
